src/aktualizr_repo/repo.cc

```diff
@@ -9,11 +9,24 @@ Repo::Repo(boost::filesystem::path path, const std::string &expires) : path_(std
 
 Json::Value Repo::signTuf(const std::string &repo_type, const Json::Value &json) {
   std::string private_key = Utils::readFile(path_ / "keys" / repo_type / "private.key");
+  std::istringstream key_type_str{Utils::readFile(path_ / "keys" / repo_type / "key_type")};
+  KeyType key_type;
+  key_type_str >> key_type;
 
-  std::string b64sig =
-      Utils::toBase64(Crypto::Sign(KeyType::kRSA2048, nullptr, private_key, Json::FastWriter().write(json)));
+  std::string b64sig = Utils::toBase64(Crypto::Sign(key_type, nullptr, private_key, Json::FastWriter().write(json)));
   Json::Value signature;
-  signature["method"] = "rsassa-pss";
+  switch (key_type) {
+    case KeyType::kRSA2048:
+    case KeyType::kRSA3072:
+    case KeyType::kRSA4096:
+      signature["method"] = "rsassa-pss";
+      break;
+    case KeyType::kED25519:
+      signature["method"] = "ed25519";
+      break;
+    default:
+      throw std::runtime_error("Unknown key type");
+  }
```

Comment:
This looks just fine for now, but at what point is it worth having a `isRsa` helper function?